managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1678,7 +1678,6 @@ public ManagedLedgerInterceptor getManagedLedgerInterceptor() {
     void clearPendingAddEntries(ManagedLedgerException e) {
         while (!pendingAddEntries.isEmpty()) {
             OpAddEntry op = pendingAddEntries.poll();
-            op.close();
```

Comment:
@codelipenghui - can you explain why you removed this line? Without it, when the `addComplete` callback is run, the method won't exit early because the state for the `OpAddEntry` will still be `INITIATED`.

Reply:
After thinking about this a little more, I think we'll need a more nuanced solution than just re-adding `op.close();` on this line. I have to sign off for now. I'll follow up with more commentary tomorrow.

Reply:
@michaeljmarshall we will not reuse the failed OpAddEntry, so we don't need to update the state of the OpAddEntry here.

Reply:
@codelipenghui - okay. My concern was that if we call `clearPendingEntries` before the BK callback completes, we would want the state to be closed so that the callback doesn't run further.